kernel/src/engine/default/json.rs

```diff
@@ -355,7 +359,8 @@ mod tests {
         async fn put_multipart_opts(
             &self,
             location: &Path,
-            opts: PutMultipartOptions,
+            #[cfg(not(feature = "arrow-55"))] opts: PutMultipartOpts,
+            #[cfg(feature = "arrow-55")] opts: PutMultipartOptions,
```

Comment:
nit: can we create a type alias instead of putting this in the params?
```rust
#[cfg(not(feature = "arrow-55"))]
type PutOpts = PutMultipartOpts;

#[cfg(feature = "arrow-55")]
type PutOpts = PutMultipartOptions;
```

Reply:
good idea but i left this inline as it's only test code and I'm trying to keep things simple :)